### PR TITLE
Fix atomic exhaustive tests.

### PR DESCRIPTION
**Pull request by rchen20**
# Summary

- This PR is a bugfix.
- It does the following:
  - Fixes part of https://github.com/LLNL/RAJA/issues/1017. 
  - Removes non-existent OMP policies.
  - Adds CAMP OpenMP Target flag automatically when RAJA enables them.

### CODE REVIEW

File: test/include/RAJA_test-forall-execpol.hpp [hunk 1]
```diff
@@ -81,8 +81,6 @@ using OpenMPForallAtomicExecPols =
 #if defined(RAJA_TEST_EXHAUSTIVE)
               , RAJA::omp_parallel_for_static_exec< >
               , RAJA::omp_parallel_for_static_exec<4>
-              , RAJA::omp_parallel_for_nowait_static_exec< >
```
Comment (rhornung67) [inline test/include/RAJA_test-forall-execpol.hpp:84]: why are these exec policies removed from the list?

Comment (rchen20) [reply test/include/RAJA_test-forall-execpol.hpp:84]: For some reason, they no longer exist as valid policies.

Comment (rchen20) [reply test/include/RAJA_test-forall-execpol.hpp:84]: Actually, these tests are already covered in lines 58-59, and in the `forall/region` tests. We can safely remove these two lines without further additions.

Comment (rchen20) [reply test/include/RAJA_test-forall-execpol.hpp:84]: Forgot these were for atomic policies, so added them back in. Tested with nvcc+gcc.

File: test/unit/reducer/tests/test-reducer-reset.hpp [hunk 2]
```diff
@@ -141,8 +141,8 @@ void testReducerReset()
   reduce_max.reset(resetVal[0]);
   reduce_minloc.reset(resetVal[0]);
   reduce_maxloc.reset(resetVal[0]);
-  reduce_minloctup.reset(resetVal[0]);
   reduce_maxloctup.reset(resetVal[0]);
```
Comment (rchen20) [inline test/unit/reducer/tests/test-reducer-reset.hpp:144]: @trws Apparently, the `maxloctup.reset()` was getting optimized out in the Sequential version of this test when compiled with NVCC + GCC. Changing the ordering fixes this for some reason, but doing so feels clunky. If there is some better way to get around this, please let me know.

Comment (rhornung67) [reply test/unit/reducer/tests/test-reducer-reset.hpp:144]: Does it get optimized away with GCC only?

Comment (rchen20) [reply test/unit/reducer/tests/test-reducer-reset.hpp:144]: Right, with GCC only. Not getting this error with other compilers.

**Review by rhornung67 — APPROVED** [review]

File: CMakeLists.txt [hunk 3]
```diff
@@ -253,6 +253,14 @@ camp source repository")
       add_subdirectory(tpl/camp)
     endif()
   endif (EXTERNAL_CAMP_SOURCE_DIR)
+
+  if (RAJA_ENABLE_TARGET_OPENMP)
```
Comment (rhornung67) [inline CMakeLists.txt:257]: @davidbeckingsale are you OK with these changes?

Comment (davidbeckingsale) [reply CMakeLists.txt:257]: Yup, seems reasonable - matches what we do in Umpire to ensure correct defines are passed to camp.

**Review by davidbeckingsale — DISMISSED** [review]

**Review by rhornung67 — APPROVED** [review]

File: test/unit/reducer/tests/test-reducer-reset.hpp (hunk 2 above)
Comment (trws) [reply test/unit/reducer/tests/test-reducer-reset.hpp:144]: When you say "optimized out" do you mean the side-effect didn't take effect so it was failing in the assertion below?  That's the kind of thing that really shouldn't happen unless we accidentally have UB somewhere.

Comment (rchen20) [reply test/unit/reducer/tests/test-reducer-reset.hpp:144]: Right, the check on line 157 fails. It returns the initialized values, or more specifically, the default initialized values.

Comment (trws) [reply test/unit/reducer/tests/test-reducer-reset.hpp:144]: That's a definite cause for concern.  It probably shouldn't matter for sequential anyway, but is it valid to use reset after compute but before get like this? I know we have places where get actually completes the reduction, and can't remember if all of that is wired up right to allow this order.

Comment (MrBurmark) [reply test/unit/reducer/tests/test-reducer-reset.hpp:144]: With the cuda and hip backends this should be safe, as reset calls finalizes the reduction by calling get before resetting the value.

Comment (rchen20) [reply test/unit/reducer/tests/test-reducer-reset.hpp:144]: Other than the author of the test not knowing about calling `get` before `reset`, there is probably no reason for this to happen. 😃

I'll modify the PR to put a `get` call into `reset()`.

Comment (rchen20) [reply test/unit/reducer/tests/test-reducer-reset.hpp:144]: That's odd, inserting `get` into the `reset` call does not help. Nor does manually calling get in the test itself. I'll still push the changes to `reset`, but keep the test with this modified ordering of calls.

**Comment by trws** [comment]
This is one you should file an issue for so we have a record of it.  
Something is fishy here.  Maybe something we should point `ubsan` toward 
and see what happens.

**Comment by rchen20** [comment]
> This is one you should file an issue for so we have a record of it. Something is fishy here. Maybe something we should point `ubsan` toward and see what happens.

Good idea, created one here https://github.com/LLNL/RAJA/issues/1217.

**Review by rhornung67 — APPROVED** [review]

**Pull request merged**